### PR TITLE
Updated the notebook manifests with 2024b image reference

### PR DESCRIPTION
**Pull request by dibryant**
Fixes for https://issues.redhat.com/browse/RHOAIENG-12629

Updated the notebook manifests with 2024b and 2024a image references

### CODE REVIEW

**Comment by atheo89** [comment]
Hi Diamond, I have two comments:
- Please review and incorporate the changes from the upstream PR: https://github.com/red-hat-data-services/notebooks/pull/389.
- We'll need to adjust all the tag versions by shifting them down and introducing the new version, 24b, as N. This change will need to be reflected across all the related files (ImageStreams, check-params-env.sh, params., commits., kustomization.yml). Please take extra care during this process, as it's easy to accidentally cause misalignments.

    - N -    2024.2
    - N-1 - 2024.1
    - N-2 - 2023.2
    - N-3 - 2023.1
    - N-4 - 1.2

**Comment by harshad16** [comment]
/lgtm
/approve

There are some bits, which needs to be adjust, they would followed up in next set of PR's.
Example: 
- verifying the annotation

**Comment by harshad16** [comment]
Merging to move along the process.

File: ci/check-params-env.sh [hunk 1]
```diff
@@ -91,143 +91,178 @@ function check_image_variable_matches_name_and_commitref() {
     local expected_build_name  # Why some of the images has `-amd64` suffix and others not?
     case "${image_variable}" in
         odh-minimal-notebook-image-n)
+            expected_name="odh-notebook-jupyter-minimal-ubi9-python-3.11"
+            expected_commitref="release-2024b"
+            expected_build_name="jupyter-minimal-ubi9-python-3.9-amd64"
```
Comment (jstourac) [inline ci/check-params-env.sh:96]: these aren't correct, similar to other occurrences - though, the CI failed so it's clear and should be fixed in the followup then

**Pull request merged**